ossapi/ossapiv2.py

```diff
@@ -195,20 +197,33 @@ def _save_token(self, token):
         with open(self.AUTHORIZATION_TOKEN_FILE, "wb+") as f:
             pickle.dump(token, f)
 
-    def _get(self, type_, url, params={}):
+    def _get(self, type_, url, params=None):
+        params = params or {}
         params = self._format_params(params)
         r = self.session.get(f"{self.BASE_URL}{url}", params=params)
-        self.log.info(f"made request to {r.request.url}")
+        self.log.info(f"made GET request to {r.request.url}")
         json_ = r.json()
         self.log.debug(f"received json: \n{json.dumps(json_, indent=4)}")
         # TODO this should just be ``if "error" in json``, but for some reason
         # ``self.search_beatmaps`` always returns an error in the response...
         # open an issue on osu-web?
-        if len(json_) == 1 and "error" in json_:
-            raise ValueError(f"api returned an error of `{json_['error']}` for "
-                f"a request to {unquote(r.request.url)}")
+        self._check_json(json_, r.request.url)
+        return self._instantiate_type(type_, json_)
+
+    def _post(self, type_, url: str, data: Optional[dict] = None):
```

Comment:
personally I don't like type hinting private methods, especially when it's obvious what the type is, like `url`. Just feels harder to read.